src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1314,6 +1314,15 @@
   <data name="PreferTypedStringBuilderAppendOverloadsRemoveToString" xml:space="preserve">
     <value>Remove the ToString call</value>
   </data>
+  <data name="PreferStringContainsOverIndexOfDescription" xml:space="preserve">
+    <value>Calls to 'String.IndexOf' where the result is compared to 0 or -1 can be replaced by 'String.Contains'</value>
```

Comment:
> where the result is compared to 0

This isn't always true, e.g. an `== 0` or `> 0` check might not be replaceable.

Reply:
Yeah, the implementation is actually much more stricter then the wordings used here.

```csharp
                    static bool CheckOperatorKindAndOperand(IBinaryOperation binaryOperation, IOperation otherOperand)
                    {
                        var operatorKind = binaryOperation.OperatorKind;
                        if (otherOperand.ConstantValue.HasValue && otherOperand.ConstantValue.Value is int intValue)
                        {
                            if ((operatorKind == BinaryOperatorKind.Equals && intValue < 0) ||
                                (operatorKind == BinaryOperatorKind.GreaterThanOrEqual && intValue == 0))
                            {
                                // This is the only case we are targeting in this analyzer
                                return true;
                            }
                        }
                        return false;
                    }
```

The wording should be adjusted to be more generic like say `Calls to 'String.IndexOf' which are only used to check the presence or absence of a sub-string within a string can be replaced by 'String.Contains'`. Not sure if this is still the best wording, but we should indicate intent instead of implementation details in this string.